sporkfish/endgame_tablebase.py

```diff
@@ -130,17 +133,26 @@ def query(self, board: Board) -> Optional[chess.Move]:
                 # This is a small performace hit but is miniscule compared to searching
                 cboard = chess.Board()
                 cboard.set_fen(board.fen())
-                for move in cboard.legal_moves:
+                dtz_scores = []
+                moves = list(cboard.legal_moves)
+                for move in moves:
                     cboard.push(move)
-                    # WDL Score: returns 2 if the side to move is winning, 0 if the position is a draw and -2 if the side to move is losing.
-                    # Returns 1 in case of a cursed win and -1 in case of a blessed loss.
-                    # Mate can be forced but the position can be drawn due to the fifty-move rule.
-                    # We check WDL score < 0 here, since we pushed our legal move and so its the opponents turn.
+                    # Returns a positive value if the side to move is winning, 0 if the position is a draw,
+                    # and a negative value if the side to move is losing.
+                    # DTZ = Number of moves to mate.
+                    # We check DTZ score < 0 here, since we pushed our legal move and so its the opponents turn.
                     # If they are losing, we are winning.
-                    if self._db.probe_wdl(cboard) < 0:
-                        return move
+                    dtz_score = self._db.probe_dtz(cboard)
```

Comment:
On second thought this can just return None in the function here if dtz_score is None - I think this is guaranteed to return an int if the position corresponds to the database name